frontend/src/Components/Form.js

```diff
@@ -0,0 +1,161 @@
+      headers: { "Content-Type": "application/json" },
+    })
+      .then(res => {
+        if (res.ok) {
+          return res.json();
+        } //eslint-disable-next-line
+        throw "Could not create user"; // we want to display error message from backend here
+      })
+      .then(json => {
+        dispatch(
+          user.actions.setAccessToken({ accessToken: json.accessToken })
+        );
+        dispatch(user.actions.setUserId({ userId: json.userId }));
+        localStorage.setItem("accessToken", json.accessToken);
+        setUsername("");
+        setPassword("");
+      })
+      .catch(err => {
+        dispatch(user.actions.setErrorMessage({ errorMessage: err }));
+      });
+  };
+
+  const loginUser = event => {
+    event.preventDefault();
+    // const LOGIN_URL = "http://localhost:8080/sessions";
+    const LOGIN_URL = "https://user-authorisation.herokuapp.com/sessions";
+    fetch(LOGIN_URL, {
+      method: "POST",
+      body: JSON.stringify({ username, password }),
+      headers: { "Content-Type": "application/json" },
+    })
+      .then(res => {
+        if (res.ok) {
+          return res.json();
+        } //eslint-disable-next-line
+        throw "Unable to sign in - Please ensure username and password are correct";
+      })
+      .then(json => {
+        dispatch(
+          user.actions.setAccessToken({ accessToken: json.accessToken })
+        );
+        dispatch(user.actions.setUserId({ userId: json.userId }));
+        localStorage.setItem("accessToken", json.accessToken);
+        setUsername("");
+        setPassword("");
+      })
+      .catch(err => {
+        dispatch(user.actions.setErrorMessage({ errorMessage: err }));
+      });
+  };
+
+  const revealSecret = event => {
+    event.preventDefault();
+    // const SECRETS_URL = "http://localhost:8080/secrets";
+    const SECRETS_URL = "https://user-authorisation.herokuapp.com/secrets";
+    fetch(SECRETS_URL, {
+      method: "GET",
+      headers: { Authorization: accessToken },
+    })
+      .then(res => {
+        if (res.ok) {
+          return res.json();
+        } //eslint-disable-next-line
+        throw "Could not get information. Make sure you are logged in and try again.";
+      })
+      .then(json => {
+        dispatch(
+          user.actions.setSecretMessage({
+            secretMessage: JSON.stringify(json.secretMessage),
+          })
+        );
+      })
+      .catch(err => {
+        dispatch(user.actions.setErrorMessage({ errorMessage: err }));
+      });
+  };
+
+  const signOut = () => {
+    dispatch(user.actions.logOut());
+  };
+
+  if (!accessToken) {
+    return (
+      <Wrapper>
+        <H1>Create account</H1>
+        <StyledForm>
+          <Label>
+            {" "}
+            {labelHeading}
+            <Input
+              value={username}
+              onChange={event => setUsername(event.target.value)}
+              type="text"
+              required
+            />
```

Comment:
For validation in both backend and frontend you could also in frontend display minlength, to prevent the possibility to create/sign in even though I have not successfully fulfilled the requirements. 

And then you could disable the button if the requirements is not met.